### PR TITLE
Signature Help for function applications

### PR DESCRIPTION
**Pull request by cartermp**
fixes https://github.com/dotnet/fsharp/issues/10302

sig-help-2

Working checklist:

- [x] Tests
- [x] Bail out when all args are applied via a pipeline (e.g. `x |> id`)
- [x] Adjust layout of tooltip when doing a pipeline
- [x] Support pipelines (e.g., `[1..10] |> List.map |`)
- [x] Support triggering inside function calls (e.g., `f (g a b) 2`)
- [x] Don't trigger on functions that are passed as params
- [x] Qualified applications (e.g., `M.f x`)
- [x] Support continued typing in methods
- [x] Try to support continued typing of identifiers in function applications
- [x] Ensure space can still be used inside tupled argument calls
- [x] Support curried F#-defined methods
- [x] Match up parameters correctly (e.g., 2nd param in 3-parameter function application)

### CODE REVIEW

File: vsintegration/src/FSharp.Editor/Completion/SignatureHelp.fs [hunk 1]
```diff
@@ -53,235 +149,488 @@ type internal FSharpSignatureHelpProvider
...
+            checkFileResults: FSharpCheckFileResults,
+            documentId: DocumentId,
+            defines: string list,
+            documentationBuilder: IDocumentationBuilder,
+            sourceText: SourceText,
+            caretPosition: int,
+            filePath: string
+        ) =
+        asyncMaybe {
+            let adjustedColumnInSource =
```
Comment (cartermp) [inline vsintegration/src/FSharp.Editor/Completion/SignatureHelp.fs:331]: Maybe don't adjust if already in App?

**Review by cartermp — COMMENTED** [review]
Yeet

**Comment by cartermp** [comment]
Alright, this is ready for review subject to tests passing.

**Review by KevinRansom — APPROVED** [review]

**Review by TIHan — APPROVED** [review]
Tests are amazing. Based on them, I feel confident with feature.

**Pull request merged**